src/components/scene/spectator.js

```diff
@@ -0,0 +1,40 @@
+var register = require('../../core/component').registerComponent;
+
+var bind = require('../../utils/').bind;
+var constants = require('../../constants/');
+var DEFAULT_SPECTATOR_CAMERA_ATTR = 'data-aframe-default-spectator-camera';
+
+/**
+ * spectator component.
+ * It injects a spectator camera if there's not one already defined.
+ */
+module.exports.Component = register('spectator', {
+  init: function () {
+    // Wait for all entities to fully load before checking for existence of camera.
+    // Since entities wait for <a-assets> to load, any cameras attaching to the scene
+    // will do so asynchronously.
+    if (this.el.hasLoaded) {
+      this.setupDefaultSpectatorCamera();
+      return;
+    }
+    this.el.addEventListener('renderstart', bind(this.setupDefaultSpectatorCamera, this));
+  },
+
+  setupDefaultSpectatorCamera: function () {
+    var sceneEl = this.el;
+    var spectatorCameraEl = sceneEl.systems.camera.spectatorCameraEl;
+    if (spectatorCameraEl) { return; }
+    spectatorCameraEl = this.defaultSpectatorCameraEl = document.createElement('a-entity');
+    spectatorCameraEl.setAttribute(DEFAULT_SPECTATOR_CAMERA_ATTR, '');
+    spectatorCameraEl.setAttribute(constants.AFRAME_INJECTED, '');
+    spectatorCameraEl.setAttribute('camera', {spectator: true});
+    spectatorCameraEl.setAttribute('wasd-controls', '');
+    spectatorCameraEl.setAttribute('look-controls', {hmdEnabled: false});
```

Comment:
i think an ideal default would be orbit controls versus the dreaded WASD + mousedrag

Reply:
Orbit controls are more appropriate when you want to focus the attention on a specific subject in the scene. I don't have any major issues with WASD + mousedrag.

Reply:
If you develop using WASD + mousedrag for two years, it's easy to get used to it. The specific subject would be the play space.

Reply:
Though wouldn't matter if we can remove the wrapper and leave it to the dev to specify controls. You can do `<a-spectator-camera>` to default the properties

Reply:
It does not matter if we remove the default. However, when implementing the viewer of A-Painter orbit controls I realized that are not very useful unless the subject is exactly on the center of the orbit controls. More often than not you want to let the user explore freely and orbit controls feel then limited. WASD + mousedrag are much more appropriate as a general solution and it is something that anybody that has played a video game is familiar with.

Reply:
We can default to wasd:drag (but use a primitive not a wrapper component), but in practice I'd use orbit.

Games don't use mouse drag though, they use pointer lock.

And we overestimate the number of people that have played a first person game with WASD. I've put aframe in front of people and they have trouble with controls.

And I think orbit controls are less limited. The head height is not fixed, you can go under or over and fly. You can use with one hand. You can have a view of the whole scene or focus on one area. We did spectator mode for multiuser a-painter with Hayden, it worked okay.